libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendMessagingExtensionConfigQuerySettingUrlResponse.ts

```diff
@@ -0,0 +1,105 @@
+ * Licensed under the MIT License.
+ */
+
+import {
+    BoolExpression,
+    BoolExpressionConverter,
+    Expression,
+    StringExpression,
+    StringExpressionConverter,
+} from 'adaptive-expressions';
+import { ActionTypes, CardAction, MessagingExtensionResponse, StringUtils } from 'botbuilder';
+import { Converter, ConverterFactory, DialogConfiguration, DialogContext, DialogTurnResult } from 'botbuilder-dialogs';
+import { BaseTeamsCacheInfoResponseDialog } from './baseTeamsCacheInfoResponseDialog';
+
+export interface SendMessagingExtensionConfigQuerySettingUrlResponseConfiguration extends DialogConfiguration {
+    disabled?: boolean | string | BoolExpression;
+    configUrl?: string | Expression | StringExpression;
+}
+
+/**
+ * Send a messaging extension 'config' response. This is the type of response used for a 'composeExtension/querySettingUrl' request.
+ */
+export class SendMessagingExtensionConfigQuerySettingUrlResponse
+    extends BaseTeamsCacheInfoResponseDialog
+    implements SendMessagingExtensionConfigQuerySettingUrlResponseConfiguration {
+    /**
+     * Class identifier.
+     */
+    public static $kind = 'Teams.SendMessagingExtensionConfigQuerySettingUrlResponse';
+
+    /**
+     * Gets or sets config url response to send.
+     *
+     * @example
+     * `${config.siteUrl}/searchSettings.html?settings=${escapedSettings}`.
+     */
+    public configUrl?: StringExpression;
+
+    public getConverter(
+        property: keyof SendMessagingExtensionConfigQuerySettingUrlResponseConfiguration
+    ): Converter | ConverterFactory {
+        switch (property) {
+            case 'disabled':
+                return new BoolExpressionConverter();
+            case 'configUrl':
+                return new StringExpressionConverter();
+            default:
+                return super.getConverter(property);
+        }
+    }
+
+    /**
+     * Called when the dialog is started and pushed onto the dialog stack.
+     *
+     * @param {DialogContext} dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param {object} options Optional, initial information to pass to the dialog.
+     * @returns {Promise<DialogTurnResult>} A promise representing the asynchronous operation.
+     */
+    public async beginDialog(dc: DialogContext, options?: Record<string, unknown>): Promise<DialogTurnResult> {
+        if (this.disabled?.getValue(dc.state)) {
+            return dc.endDialog();
+        }
+
+        const configUrl = this.configUrl?.getValue(dc.state);
+        if (!configUrl) {
+            throw new Error(`ConfigUrl is required for ${SendMessagingExtensionConfigQuerySettingUrlResponse.$kind}.`);
+        }
+
+        const response: MessagingExtensionResponse = {
+            composeExtension: {
+                type: 'config',
+                suggestedActions: {
+                    actions: [
+                        <CardAction>{
```

Comment:
`CardAction.title` is a required property in JS, but not .NET (because of how JSON works in .NET, I think). It seems weird to add `title: ''`. Discussing with Eric, now, about whether or not we should add a `title` property to `...ConfigQuerySettingUrlResponse`.

Reply:
ended up just hardcoding it as an empty string. It's never seen by the user...just used to have Teams fetch a URL.